src/Command/Pull/PullCommandBase.php

```diff
@@ -507,14 +507,21 @@ private function promptChooseDatabases(
   }
 
   protected function runComposerScripts(callable $outputCallback = NULL): void {
-    if (file_exists($this->dir . '/composer.json') && $this->localMachineHelper->commandExists('composer')) {
-      $this->checklist->addItem("Installing Composer dependencies");
-      $this->composerInstall($outputCallback);
-      $this->checklist->completePreviousItem();
+    if (!file_exists(Path::join($this->dir, 'composer.json'))) {
+      $this->logger->notice('composer.json file not found. Skipping composer install.');
+      return;
     }
-    else {
-      $this->logger->notice('composer or composer.json file not found. Skipping composer install.');
+    if (!$this->localMachineHelper->commandExists('composer')) {
+      $this->logger->notice('Composer not found. Skipping composer install.');
+      return;
     }
+    if (file_exists(Path::join($this->dir, 'vendor'))) {
+      $this->logger->notice('Composer dependencies already installed. Skipping composer install.');
+      return;
+    }
+    $this->checklist->addItem("Installing Composer dependencies");
+    $this->composerInstall($outputCallback);
+    $this->checklist->completePreviousItem();
```

Comment:
Not sure if the mutation tests are catching something here or if they just need to be updated. It seems it's the latter.

Reply:
Due to idiosyncrasies with how the Symfony Command Tester class handles output, our test cases can't verify that these checklist items appear in the command output. Hence the escaped mutant. But it's not caused by this PR, so we can ignore it.